backend/sozisel/test/support/factory.ex

```diff
@@ -12,4 +16,34 @@ defmodule Sozisel.Factory do
       password_hash: attrs[:password_hash] || Bcrypt.hash_pwd_salt(attrs[:password] || "password")
     }
   end
+
+  def template_factory(attrs) do
+    %Template{
+      deleted_at: attrs[:deleted_at] || nil,
+      estimated_time: attrs[:estimated_time] || sequence(:estimated_time, & &1),
+      is_abstract: attrs[:is_abstract] || sequence(:is_abstract, [false, true]),
+      is_public: attrs[:is_public] || sequence(:is_public, [false, true]),
+      name: attrs[:name] || sequence(:name, &"Template name no. #{&1}"),
+      user_id: attrs[:user_id] || build(:user).id
+    }
+  end
+
+  def agenda_entry_factory(attrs) do
+    %AgendaEntry{
+      start_minute: attrs[:start_minute] || sequence(:start_minute, & &1),
+      name: attrs[:name] || sequence(:name, &"Agenda entry name no. #{&1}"),
+      session_template_id: attrs[:session_template_id] || build(:template).id
+    }
+  end
+
+  def session_factory(attrs) do
+    %Session{
+      entry_password: attrs[:entry_password] || sequence(:entry_password, &"some_password-#{&1}"),
+      start_time: attrs[:start_time] || "2011-05-18T15:01:01.000000Z",
+      use_jitsi: attrs[:use_jitsi] || sequence(:use_jitsi, [false, true]),
+      name: attrs[:name] || sequence(:name, &"Session name no. #{&1}"),
+      user_id: attrs[:user_id] || build(:user).id,
```

Comment:
This won't work. If you perform `insert(:session)` then the user may not be inserted into the database therefore this will fail. Use something like this:
```elixir
user: attrs[:user] || build(:user)
```
Then the ex_machina will be able to perform insert of given user (but check it beforehand as I don't remember correctly).

Reply:
At first I did exactly the same as you suggest, however then assertion fails, cause session created by ex_machina has loaded user, and session queried from database doesn't (we don't use preload).

Reply:
If this works then it's fine I guess.

Reply:
@Qizot Tests are passing, however you were right. I've checked it and user is not inserted into the database in this case. I solved it by changing `build(:user).id` into `insert(:user).id`. Now it works fine and user is inserted into the database.